alacritty_terminal/src/event.rs

```diff
@@ -52,6 +52,9 @@ pub enum Event {
 
     /// Shutdown request.
     Exit,
+
+    /// Child process exited with an error code.
+    ChildExit(i32),
```

Comment:
Just to make sure: You don't care about children that get killed? Since those obviously won't have an exit code.

Reply:
I do not think so, but my use cases might be not very mainstream (judging by the fact that I have created 3 feature requests related to them), so ready to adjust the logic the way it seems more appropriate.

All my 3 cases are related to Zed and this part of code: https://github.com/zed-industries/zed/blob/1081ba7a62ed67ba7dae0e2304842e82a589f25e/crates/terminal/src/terminal.rs#L322-L338

* interactive shell session

Does not really matter how and when children exit: `Event::Exit` is used for hiding corresponding UI elements, and it will be sent anyway, no dependence on the child being killed or not.

* (future) collab terminal with tty bytes sent over the server into another instance

Either no children will be used at all, or, if we'll come up with a magic wrapper to make output easier, its child exit should be ignored as not something exposed to the user => no need to know if it's killed too

* running bash-like commands that user defined in Zed elsewhere

Here is what it's all about: those commands spawn a dedicated, task tabs that will only show that command's output — I would want to e.g. change the terminal tab icon based on various child exit statuses, but right now all Alacritty has for shutdown reporting is `Event::Exit`.

If we can rely on Alacritty always sending `Event::ChildExit` on such command finish (which seems to work, based on my tests), killed and all other non-natural abrupts can be derived from the fact that we've already received `Event::Exit`but no `Event::ChildExit` was sent before.

Works for me and does not seem like we need to know about the killing here too.